src/main/c/cext/globals.c

```diff
@@ -48,6 +52,18 @@ void rb_define_variable(const char *name, VALUE *var) {
   rb_define_hooked_variable(name, var, 0, 0);
 }
 
+void rb_define_virtual_variable(const char *name, rb_gvar_getter_t *getter, rb_gvar_setter_t *setter) {
+    if (!getter) {
+      getter = rb_gvar_val_getter;
+    }
+
+    if (!setter) {
+      setter = rb_gvar_readonly_setter;
+    }
+
+    rb_define_hooked_variable(name, 0, getter, setter);
```

Comment:
Won't this always return 0=Qfalse until set explicitly? What's the semantics of `rb_define_virtual_variable()`? Specs would be good for it.

Reply:
I took the code exactly from MRI, only changing the single line branches to have braces like we've done elsewhere:

https://github.com/ruby/ruby/blob/v3_0_4/variable.c#L592-L601

If there's a flaw in the logic, it exists upstream as well. The entire implementation is in C. I.e., I didn't add any Ruby or Java support code for it.